server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -280,14 +279,22 @@ def _remove_empty_elements(data: Any) -> Any:
                 if (cleaned := _remove_empty_elements(v)) is not None:
                     data_transformed[k] = cleaned
 
-        return data_transformed or None
+        if isinstance(data_transformed, list):
+            return data_transformed
+        else:
+            return data_transformed or None
+
     elif isinstance(data, list):
         data_transformed = [
             cleaned
             for item in data
             if (cleaned := _remove_empty_elements(item)) is not None  # type: ignore[assignment]
         ]
-        return data_transformed or None
+
+        if isinstance(data_transformed, list):
+            return data_transformed
+        else:
+            return data_transformed or None
```

Comment:
The `isinstance` check is redundant. The comprehension could be returned directly here

Reply:
ah yes, true. updated here : https://github.com/ToucanToco/weaverbird/pull/1838/commits/75bdda95